build.gradle

```diff
@@ -77,6 +77,8 @@ test {
     useJUnitPlatform()
 }
 
+sourceSets.main.resources.srcDirs(sourceSets.test.resources.srcDirs) // include test books
```

Comment:
I'm confused. I thought you were going to move the test data to the test sourceset, then remove the ad-hoc exclude directives from the jar task below. Not sure what this change is supposed to do.

Reply:
[The way directories are scanned](https://github.com/Vazkii/Patchouli/blob/14fcf637febf19224b3eaa7485f27f203fa80ad1/src/main/java/vazkii/patchouli/common/book/BookRegistry.java#L122) ([this](https://github.com/Vazkii/Patchouli/blob/82e38d5e15f3968ec9774b708b7552ead15a0661/src/main/java/vazkii/patchouli/common/book/BookRegistry.java#L143) on Fabric) requires that book resources are in the same directory as the mod metadata file `mods.toml`/`fabric.mod.json`.

So, without code changes, the test resources have to be in the resource dir that the mod metadata file is in.

Some options are:
- Add test source set resources to the main source set. This still requires the resources to be excluded from the final jar. (current)
- Make a new source set, include test and main classes and resources, and run from that. Possible, but slightly hacky and loader dependent.
- Make code changes so books can be scanned in additional directories. Possibly by environmental variables or such.

Reply:
or you can make the test source set depend on main and run with the test source set when you want the test data

Reply:
I don't see much benefit of either of the other approaches compared to what we have now, which is a bit of a hassle to exclude at build time but it works.

Especially now that we have RP-based books, most of our test books should probably be moved there anyways.